lib/vnet/daemon/client_darwin.h

```diff
@@ -0,0 +1,53 @@
+#ifndef client_darwin_h
+#define client_darwin_h
+
+#import <Foundation/Foundation.h>
+
+typedef struct BundlePathResult {
+    const char * bundlePath;
+} BundlePathResult;
+
+// BundlePath updates bundlePath of result with the path
+// to the bundle directory that contains the current executable.
+// bundlePath is an empty string if the bundle details could not be fetched.
+// It might return a path even for executables that are not in a bundle.
+// In that case, calling codesign --verify on that path will simply return with 1.
+void BundlePath(struct BundlePathResult *result);
```

Comment:
Suggestion: use the typedef declared above, explicitly mark output params:

```suggestion
void BundlePath(BundlePathResult *outResult);
```

Reply:
Could you tell me or give me a quick link to something that explains the difference between `void BundlePath(struct BundlePathResult *result)` and `void BundlePath(BundlePathResult *result)`? I wasn't aware of the latter.

Reply:
Sure.

My poor man's explanation is this: structs in C have their own "namespace", so when you declare "struct Foo", its "type" is always "struct Foo" (instead of just "Foo", which is what we've come to expect from other languages).

typedefs are a way to give a nicer, concise name to complex things. Because typing "struct Foo" everywhere is cumbersome, and forces your program to know that Foo is a struct, it's common to always typedef structs to a "nicer" name so we can drop the "struct" part from the syntax.

Usually this means we typedef "struct Foo" to "Foo" and get own with our lives. In some codebases you'll see "struct _Foo" typedef as "Foo" as well.

- https://stackoverflow.com/a/2566035